src/features/student/declaredPrograms/components/interactions/formFields/DeclaredProgramInstitutionFormField/DeclaredProgramInstitutionFormField.test.ts

```diff
@@ -0,0 +1,182 @@
+import type { AddDeclaredProgramForm, DeclaredProgramFormData } from '@/features/student/declaredPrograms/types/forms.types'
+import DeclaredProgramInstitutionFormField from '@/features/student/declaredPrograms/components/interactions/formFields/DeclaredProgramInstitutionFormField/DeclaredProgramInstitutionFormField.vue'
+import { DECLARED_PROGRAM_ORGANIZATION_MAX_LENGTH } from '@/features/student/declaredPrograms/config'
+import { BddTest } from '@avenirs-esr/avenirs-dsav/test-utils'
+import { useForm } from '@tanstack/vue-form'
+import { mount, type VueWrapper } from '@vue/test-utils'
+import { beforeEach, expect, vi } from 'vitest'
+
+const DeclaredProgramInstitutionInputStub = defineComponent({
+  name: 'DeclaredProgramInstitutionInput',
+  props: {
+    modelValue: String,
+    errorMessage: String
+  },
+  emits: ['update:modelValue', 'blur'],
+  template: `
+    <div data-testid="declared-program-institution-input-stub">
+      <input
+        :value="modelValue"
+        @input="$emit('update:modelValue', $event.target.value)"
+        @blur="$emit('blur')"
+      />
+    </div>
+  `
+})
+
+const TestWrapper = defineComponent({
+  components: {
+    DeclaredProgramInstitutionFormField
+  },
+  setup () {
+    const form = useForm({
+      defaultValues: {
+        organization: ''
+      } as DeclaredProgramFormData,
+      validators: {
+        onSubmit ({ value }) {
+          if (!value.organization || value.organization.trim() === '') {
+            return {
+              fields: {
+                organization: 'L\'établissement est requis'
+              }
+            }
+          }
+          return { fields: { organization: undefined } }
+        }
+      }
+    }) as unknown as AddDeclaredProgramForm
+
+    return { form }
+  },
+  template: `
+    <form @submit.prevent="form.handleSubmit">
+      <DeclaredProgramInstitutionFormField :form="form" />
+    </form>
+  `
+})
+
+BddTest().given('a declared program institution form field', () => {
+  let wrapper: VueWrapper<InstanceType<typeof TestWrapper>>
+
+  const stubs = {
+    DeclaredProgramInstitutionInput: DeclaredProgramInstitutionInputStub
+  }
+
+  BddTest().when('the component is mounted', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should render the institution input component', () => {
+      const input = wrapper.findComponent({ name: 'DeclaredProgramInstitutionInput' })
+      expect(input.exists()).toBe(true)
+    })
+
+    BddTest().then('it should have empty initial value', () => {
+      const input = wrapper.findComponent({ name: 'DeclaredProgramInstitutionInput' })
+      expect(input.props('modelValue')).toBe('')
+    })
+  })
+
+  BddTest().and('the user enters an institution name', () => {
+    beforeEach(async () => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
```

Comment:
```suggestion
  BddTest().and('the user enters an institution name', () => {
```

Comme pour l'autre fichier de test : tous les and doivent être dans le when au dessus et tu peux enlever ce code répété à chaque fois